litex/build/gowin/apicula.py

```diff
@@ -25,16 +25,27 @@ def build_io_constraints(self):
         return (self._build_name + ".cst", "CST")
 
     def finalize(self):
+        devicename = self.platform.devicename
+        # Non-exhaustive list of family aliases that Gowin IDE supports but don't have a unique database
+        if devicename == "GW1NR-9C":
+            devicename = "GW1N-9C"
+        elif devicename == "GW1NR-9":
+            devicename = "GW1N-9"
+        elif devicename == "GW1NSR-4C" or devicename == "GW1NSR-4":
+            devicename = "GW1NS-4"
+        elif devicename == "GW1NR-4C" or devicename == "GW1NR-4":
+            devicename = "GW1N-4"
+
```

Comment:
why not using a dict as done for xilinx ?

Reply:
Because there is a many to one relationship and I doubt a dict is faster for small numbers but also because I ported it from C. I don't really have strong feelings one way or the other.